Mathlib/Tactic/ComputeDegree.lean

```diff
@@ -0,0 +1,206 @@
+/-
+Copyright (c) 2023 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+
+import Mathlib.Data.Polynomial.Degree.Definitions
+
+/-!
+
+# `compute_degree_le` a tactic for computing degrees of polynomials
+
+This file defines the tactic `compute_degree_le`.
+
+Using `compute_degree_le` when the goal is of the form `natDegree f ≤ d` or `degree f ≤ d`,
+tries to solve the goal.
+It may leave a side-goal of the form `d' ≤ d`, in case it is not entirely successful.
+
+See the doc-string for more details.
+
+##  Future work
+
+* Currently, the tactic does not handle goals of the form `degree f ≤ d`, where `d` is not the
+  coercion of a `Nat`.  This should be fixed.
+* Deal with equalities, instead of inequalities (i.e. implement `compute_degree`).
+* Add functionality to deal with (some) exponents that are not closed natural numbers.
```

Comment:
```suggestion
```

The current implementation does not assume anything from the exponents, other than that they are terms of type `Nat`.